test/e2e/validation_test.go

```diff
@@ -275,3 +275,38 @@ func createWindowsMachineConfig(namespace string, isReplicasFieldRequired bool,
 	}
 	return wmc
 }
+
+// testWorkerLabel tests if the worker label has been applied properly
+func testWorkerLabel(t *testing.T, nodeCount int) {
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup()
+	nodes, err := framework.Global.KubeClient.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: nc.WindowsOSLabel})
```

Comment:
You can use kubeclient from testCtx. And you can get nodes from the global context `gc`